File: Python/ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        map = {"{": "}","[": "]","(":")"}
        size = 0
        
        for c in s:
            if c in map:
                stack.append(c)
            elif len(stack) == 0:
                return False
            elif map[stack[-1]] != c:
                return False
            else:
                stack.pop()

        return len(stack) == 0

File: Java/ValidParentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        for (int i = 0; i < s.length(); i++) {
            int size = stack.size();
            switch (s.charAt(i)) {
                case ')':
                    if (size == 0) {
                        return false;
                    }
                    if (stack.pop() != '(') {
                        return false;
                    }
                    break;
                case '}':
                    if (size == 0) {
                        return false;
                    }
                    if (stack.pop() != '{') {
                        return false;
                    }
                    break;
                case ']':
                    if (size == 0) {
                        return false;
                    }
                    if (stack.pop() != '[') {
                        return false;
                    }   
                    break;
                default:
                    stack.push(s.charAt(i));
            }
        }
        return stack.empty();
    }
}
